### ISSUE TITLE
Is it trained after loading the pre training model or not

### ISSUE THREAD

**Issue by Arvin1201** [argusswift/YOLOv4-pytorch#2]
As described in the title

**Comment by argusswift** [comment]
It is better for you to conduct your training program after the pre training model is loaded, which will hugely improve the validation accuracy.

**Comment by Arvin1201** [comment]
> It is better for you to conduct your training program after the pre training model is loaded, which will hugely improve the validation accuracy.

I understand what you said. I mean, under what circumstances did your model train? Did you train with or without the author's weight

**Comment by argusswift** [comment]
yeah...I had used the author's weight (yolov4.weights) to get my results.For you, you can follow the instructions in read.me to conduct your experiment.

**Comment by Arvin1201** [comment]
> yeah...I had used the author's weight (yolov4.weights) to get my results.For you, you can follow the instructions in read.me to conduct your experiment.

I don't think that if you want to verify the performance of your reproduced code, you should not load the original author's weight during training
In particular, I expect you to validate your code and feed back your test results without loading the original author weight file.
Thanks.

**Comment by Arvin1201** [comment]
> > > yeah...I had used the author's weight (yolov4.weights) to get my results.For you, you can follow the instructions in read.me to conduct your experiment.
> > 
> >
> > I don't think that if you want to verify the performance of your reproduced code, you should not load the original author's weight during training
> > In particular, I expect you to validate your code and feed back your test results without loading the original author weight file.
> > Thanks.
> 
> yeah,you are excellent. I have changed my source code to verify the results. In this section, I have already loaded the CSPDarknet-53 weight file in the feature extraction network.

哈哈,才知道你是中国人, 咨询下在不加载预训练模型情况下进行训练时,你的代码训练的效果是多少???

**Comment by Arvin1201** [comment]
> > > yeah...I had used the author's weight (yolov4.weights) to get my results.For you, you can follow the instructions in read.me to conduct your experiment.
> > 
> > 
> > I don't think that if you want to verify the performance of your reproduced code, you should not load the original author's weight during training
> > In particular, I expect you to validate your code and feed back your test results without loading the original author weight file.
> > Thanks.
> 
> yeah,you are excellent. I have changed my source code to verify the results. In this section, I have already loaded the CSPDarknet-53 weight file in the feature extraction network.

你这个是最新的代码吗????

**Comment by argusswift** [comment]
> > > > yeah...I had used the author's weight (yolov4.weights) to get my results.For you, you can follow the instructions in read.me to conduct your experiment.
> > > 
> > > 
> > > I don't think that if you want to verify the performance of your reproduced code, you should not load the original author's weight during training
> > > In particular, I expect you to validate your code and feed back your test results without loading the original author weight file.
> > > Thanks.
> > 
> > 
> > yeah,you are excellent. I have changed my source code to verify the results. In this section, I have already loaded the CSPDarknet-53 weight file in the feature extraction network.
> 
> 哈哈,才知道你是中国人, 咨询下在不加载预训练模型情况下进行训练时,你的代码训练的效果是多少???

这个没有测试过，但结果应该不会很差。

**Comment by argusswift** [comment]
> > > > yeah...I had used the author's weight (yolov4.weights) to get my results.For you, you can follow the instructions in read.me to conduct your experiment.
> > > 
> > > 
> > > I don't think that if you want to verify the performance of your reproduced code, you should not load the original author's weight during training
> > > In particular, I expect you to validate your code and feed back your test results without loading the original author weight file.
> > > Thanks.
> > 
> > 
> > yeah,you are excellent. I have changed my source code to verify the results. In this section, I have already loaded the CSPDarknet-53 weight file in the feature extraction network.
> 
> 你这个是最新的代码吗????

这个暂时是最新的，后面还有新的模块后续还会更新。

**Comment by Arvin1201** [comment]
> 没有测试过，但结果应该不

这个我进行过测试, 训练不出来模型, 所以就请教你下

**Comment by argusswift** [comment]
> > 没有测试过，但结果应该不
> 
> 这个我进行过测试, 训练不出来模型, 所以就请教你下

是用我的代码没有训练出模型？不加载预训练模型的话，还是可以训练的，不过训练会比较慢，很难拟合大型数据集并且精度会有所降低。

**Comment by Arvin1201** [comment]
我从新训练下,不加载模型在VOC2007上我训练了51个epoch,map能达到64%

**Comment by Arvin1201** [comment]
不过你的数据处理部分是没有mosaic的

**Comment by sudo-rm-covid19** [comment]
Hi @argusswift ,
I think there is a bug in the forward function of CSPStage and CSPFirstStage:
```python
        x0 = self.split_conv0(x)
        x1 = self.split_conv1(x)

        x1 = self.blocks_conv(x1)

        x = torch.cat([x0, x1], dim=1) # where [x1, x0] should be used as it is in the original implementation
        x = self.concat_conv(x)
```
Thanks.

**Comment by argusswift** [comment]
> Hi @argusswift ,
> I think there is a bug in the forward function of CSPStage and CSPFirstStage:
> 
> ```python
>         x0 = self.split_conv0(x)
>         x1 = self.split_conv1(x)
> 
>         x1 = self.blocks_conv(x1)
> 
>         x = torch.cat([x0, x1], dim=1) # where [x1, x0] should be used as it is in the original implementation
>         x = self.concat_conv(x)
> ```
> 
> Thanks.

Thanks, your are right.If possible, you can pull requests with the modified code.Thank you again!

**Comment by mercuryson** [comment]
这个提供的YOLOV4 DARKNET PRE-TRAINED WEIGHT是在COCO上训练的吗？为什么上来TOTAL LOSS非常的大。。。
